pkg/controller/masteruserrecord/sync.go

```diff
@@ -185,8 +186,11 @@ func (s *Synchronizer) alignReadiness() (bool, error) {
 	if condition.IsNotTrue(s.record.Status.Conditions, toolchainv1alpha1.MasterUserRecordUserProvisionedNotificationCreated) {
 		notification := &toolchainv1alpha1.Notification{
 			ObjectMeta: v1.ObjectMeta{
-				Name:      s.record.Name + "-provisioned",
+				Name:      s.record.Name + "-provisioned" + "-" + strconv.FormatInt(v1.Now().UnixNano(), 10),
 				Namespace: s.record.Namespace,
+				Labels: map[string]string{
+					toolchainv1alpha1.NotificationUserNameLabelKey: s.record.Name,
```

Comment:
This label is not used anywhere, is it?

Reply:
it's used in the e2e tests now. @xcoulon suggested that we add a label to look up the notifications for a specific user in the tests. without the label, we need to check if the name contains the user name and the reason we are notifying (deactivated for instance). instead, it may be better to use auto-generated names for the notification name and use the label to understand who the notification is for. perhaps we can add a label for the reason we are notifying too? what do you think?

Reply:
OK. Adding username and notification reason/type labels for lookups is probably a good idea... But:
- let's keep the overall notification name template for generated names as it is now: `GeneratedName: <username>-<type>-` so it's a human friendly name.
- let's add comments to the code specifying that the labels atm are used only for easier notification lookup during debugging and in e2e tests.

Reply:
Okay, sounds good!